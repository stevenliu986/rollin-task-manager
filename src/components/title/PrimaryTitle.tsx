import { Typography } from "antd";
import React from "react";

interface IPrimaryTitleProps extends React.ComponentProps<typeof Typography.Title> {
    children: React.ReactNode;
}

const DEFAULT_STYLE: React.CSSProperties = {
    textTransform: 'uppercase',
    fontSize: '2rem',
    fontWeight: 700,
    letterSpacing: '-0.1em',
    color: '#252525',
    margin: '0 0 1rem 0',
    fontFamily: 'inherit',
    lineHeight: 1.2,
}

export default function PrimaryTitle({
    children, style, ...props
}: Readonly<IPrimaryTitleProps>) {

    const mergedStyle = { ...DEFAULT_STYLE, ...style };
    return (
        <Typography.Title
            level={1}
            style={mergedStyle}
            {...props}
        >
            {children}
        </Typography.Title>
    );
}