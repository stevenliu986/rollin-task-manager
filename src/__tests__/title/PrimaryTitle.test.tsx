import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import PrimaryTitle from '../../components/title/PrimaryTitle';

vi.mock('antd', async (importOriginal) => {
    const mod = await importOriginal();
    return {
        ...mod,
        Typography: {
            Title: ({ children, style, ...props }: any) => (
                <h1 style={style} data-testid="antd-title" {...props}>
                    {children}
                </h1>
            )
        }
    };
});

describe('PrimaryTitle Component', () => {
    it('renders children with default styles', () => {
        render(<PrimaryTitle>Test Title</PrimaryTitle>);

        const title = screen.getByTestId('antd-title');

        expect(title).toHaveTextContent('Test Title');
        expect(title).toHaveStyle({
            textTransform: 'uppercase',
            fontSize: '2rem',
            fontWeight: '700',
            color: '#252525',
            margin: '0 0 1rem 0'
        });
    });

    it('merges custom styles with default styles', () => {
        const customStyle = {
            color: '#FF0000',
            marginBottom: '2rem'
        };

        render(
            <PrimaryTitle style={customStyle}>
                Styled Title
            </PrimaryTitle>
        );

        const title = screen.getByTestId('antd-title');

        expect(title).toHaveStyle({
            color: '#FF0000',
            marginBottom: '2rem'
        });

        expect(title).toHaveStyle({
            textTransform: 'uppercase',
            fontSize: '2rem'
        });
    });

    it('passes through additional props to Typography.Title', () => {
        render(
            <PrimaryTitle
                className="custom-class"
                id="main-title"
                aria-label="Primary Title"
            >
                Props Test
            </PrimaryTitle>
        );

        const title = screen.getByTestId('antd-title');

        expect(title).toHaveAttribute('class', 'custom-class');
        expect(title).toHaveAttribute('id', 'main-title');
        expect(title).toHaveAttribute('aria-label', 'Primary Title');
    });

    it('renders with correct heading level', () => {
        render(<PrimaryTitle level={2}>Heading Level</PrimaryTitle>);

        const title = screen.getByTestId('antd-title');
        expect(title.tagName).toBe('H1');
    });

    it('matches snapshot with default props', () => {
        const { asFragment } = render(
            <PrimaryTitle>Snapshot Test</PrimaryTitle>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot with custom styles', () => {
        const { asFragment } = render(
            <PrimaryTitle style={{ color: 'blue', fontSize: '3rem' }}>
                Styled Snapshot
            </PrimaryTitle>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});