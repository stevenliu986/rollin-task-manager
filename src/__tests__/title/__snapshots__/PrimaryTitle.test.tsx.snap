// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PrimaryTitle Component > matches snapshot with custom styles 1`] = `
<DocumentFragment>
  <h1
    data-testid="antd-title"
    level="1"
    style="text-transform: uppercase; font-size: 3rem; font-weight: 700; letter-spacing: -0.1em; color: blue; margin: 0px 0px 1rem 0px; font-family: inherit; line-height: 1.2;"
  >
    Styled Snapshot
  </h1>
</DocumentFragment>
`;

exports[`PrimaryTitle Component > matches snapshot with default props 1`] = `
<DocumentFragment>
  <h1
    data-testid="antd-title"
    level="1"
    style="text-transform: uppercase; font-size: 2rem; font-weight: 700; letter-spacing: -0.1em; color: rgb(37, 37, 37); margin: 0px 0px 1rem 0px; font-family: inherit; line-height: 1.2;"
  >
    Snapshot Test
  </h1>
</DocumentFragment>
`;
